protocol/xmrtaker/message_handler.go

```diff
@@ -81,8 +81,14 @@ func (s *swapState) setNextExpectedEvent(event EventType) error {
 }
 
 func (s *swapState) handleSendKeysMessage(msg *net.SendKeysMessage) (net.Message, error) {
-	if msg.ProvidedAmount < s.info.ReceivedAmount {
-		return nil, fmt.Errorf("receiving amount is not the same as expected: got %v, expected %v",
+
+	if msg.ProvidedAmount == nil {
+		return nil, errMissingProvidedAmount
+	}
+
+	// TODO: THIS CHECK LOOKS BACKWARDS?!?
```

Comment:
I think it's fine? we're checking that what the counterparty is claiming they will provide is the same (or greater than) what we expected them to, as calculated when the swap state was created